ilcompiler: System.Diagnostics.Debugger.IsSupported has already been added

### Description

Error on `dotnet publish -c Release` if `<DebuggerSupport>false</DebuggerSupport>` and `<PublishAot>true</PublishAot>` specified in `.csproj`
```
EXEC : error : An item with the same key has already been added. Key: System.Diagnostics.Debugger.IsSupported [WorkerService1\WorkerService1.csproj]
  System.ArgumentException: An item with the same key has already been added. Key: System.Diagnostics.Debugger.IsSupported
     at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException[T](T) + 0x14
     at System.Collections.Generic.Dictionary`2.TryInsert(TKey, TValue, InsertionBehavior) + 0x27f
     at System.Collections.Generic.Dictionary`2.AddRange(IEnumerable`1) + 0x13b
     at ILCompiler.FeatureSwitchManager..ctor(ILProvider, Logger, IEnumerable`1) + 0x37
     at ILCompiler.Program.Run() + 0xeb2
     at ILCompiler.ILCompilerRootCommand.<>c__DisplayClass203_0.<.ctor>b__0(InvocationContext context) + 0x20c

.nuget\packages\microsoft.dotnet.ilcompiler\8.0.0-preview.3.23174.8\build\Microsoft.NETCore.Native.targets(266,5): error MSB3073
```

### Reproduction Steps

<!---->

### Expected behavior

<!---->

### Actual behavior

<!---->

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

SDK:
Version:   8.0.100-preview.3.23178.7
Commit:    e300b0e1e6


### Other information

_No response_

--- a/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs
+++ b/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs
@@ -28,7 +28,13 @@ namespace ILCompiler
         public FeatureSwitchManager(ILProvider nestedILProvider, Logger logger, IEnumerable<KeyValuePair<string, bool>> switchValues)
         {
             _nestedILProvider = nestedILProvider;
-            _hashtable = new FeatureSwitchHashtable(logger, new Dictionary<string, bool>(switchValues));
+
+            // Last setting wins
+            var dictionary = new Dictionary<string, bool>();
+            foreach ((string name, bool value) in switchValues)
+                dictionary[name] = value;
+
+            _hashtable = new FeatureSwitchHashtable(logger, dictionary);
         }
 
         private BodySubstitution GetSubstitution(MethodDesc method)

